pkg/manager/internal/controllers/organization_controller.go

```diff
@@ -152,3 +172,93 @@ func (r *OrganizationReconciler) reconcileNamespace(ctx context.Context, log log
 	}
 	return nil
 }
+
+func (r *OrganizationReconciler) reconcileOrganizationRoleTemplates(ctx context.Context, organization *corev1alpha1.Organization) ([]*corev1alpha1.OrganizationRoleTemplate, error) {
+	desiredProjectAdminOrganizationRoleTemplate := r.buildProjectAdminOrganizationRoleTemplate(organization)
+	desiredRBACAdminOrganizationRoleTemplate := r.buildRBACAdminOrganizationRoleTemplate(organization)
+
+	var templates []*corev1alpha1.OrganizationRoleTemplate
+	projectTemplate, err := r.reconcileOrganizationRoleTemplate(ctx, organization, desiredProjectAdminOrganizationRoleTemplate)
+	if err != nil {
+		return nil, fmt.Errorf("reconcile project-admin OrganizationRoleTemplate: %w", err)
+	}
+	templates = append(templates, projectTemplate)
+
+	rbacTemplate, err := r.reconcileOrganizationRoleTemplate(ctx, organization, desiredRBACAdminOrganizationRoleTemplate)
+	if err != nil {
+		return nil, fmt.Errorf("reconcile rbac-admin OrganizationRoleTemplate: %w", err)
+	}
+	templates = append(templates, rbacTemplate)
+
+	return templates, nil
+}
+
+func (r *OrganizationReconciler) reconcileOrganizationRoleTemplate(ctx context.Context,
```

Comment:
This is buggy by design; You're never properly setting the owner reference on Creation (only on the update, since this would trigger the modification function from the `controllerutil` )

Reply:
Nope, that works fine.
As mentioned in their documentation: https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/controller/controllerutil/controllerutil.go#L189
`The MutateFn is called regardless of creating or updating an object.`

Reply:
wait, wat? That makes so much more sense, thank you!